File: 01 - Iniciante/1094 - Experiencias/problem1094.py
""" 
	URI Online Judge 
	Problema 1094 - Experiências
    https://www.urionlinejudge.com.br/repository/UOJ_1094.html 
"""

casos = int(input())

quantidade = {
    "rato": 0,
    "coelho": 0,
    "sapo": 0
}

percentual = {
    "rato": 0,
    "coelho": 0,
    "sapo": 0
}

total = 0

for caso in range(casos):
    qtd, tipo = input().split()

    if (tipo == 'R'):
        quantidade["rato"] += int(qtd)
    
    if (tipo == 'C'):
        quantidade["coelho"] += int(qtd)
    
    if (tipo == 'S'):
        quantidade["sapo"] += int(qtd)
    
    total += int(qtd)

percentual["rato"] = 100 * quantidade["rato"] / total
percentual["coelho"] = 100 * quantidade["coelho"] / total
percentual["sapo"] = 100 * quantidade["sapo"] / total

print("Total: %d cobaias" % total)
print("Total de coelhos: %d" % quantidade["coelho"])
print("Total de ratos: %d" % quantidade["rato"])
print("Total de sapos: %d" % quantidade["sapo"])
print("Percentual de coelhos: %.2f %%" % percentual["coelho"])
print("Percentual de ratos: %.2f %%" % percentual["rato"])
print("Percentual de sapos: %.2f %%" % percentual["sapo"])

File: 01 - Iniciante/1094 - Experiencias/problem1094.cpp
/*  
	URI Online Judge 
	Problema 1094 - Experiencias
    https://www.urionlinejudge.com.br/repository/UOJ_1080.html 
 */

#include <iostream>
#include <iomanip> 

using namespace std;

int main() {

    int casos, qtdInput, qtdSapos = 0, qtdCoelhos = 0, qtdRatos = 0, total;
    char tipo;
    
    cin >> casos;

    for (int i=0; i<casos; i++){
        cin >> qtdInput >> tipo;
        total += qtdInput;

        switch(tipo) {
            case 'R': 
                qtdRatos += qtdInput;
                break;
            case 'S': 
                qtdSapos += qtdInput;
                break;
            case 'C': 
                qtdCoelhos += qtdInput;
                break;
        }

    }

    cout << "Total: " << total << " cobaias" << endl;
    cout << "Total de coelhos: " << qtdCoelhos << endl;
    cout << "Total de ratos: " << qtdRatos << endl;
    cout << "Total de sapos: " << qtdSapos << endl;
    cout << fixed << setprecision(2);
    cout << "Percentual de coelhos: " << 100.0 * qtdCoelhos / total << " %" << endl;
    cout << "Percentual de ratos: " << 100.0 * qtdRatos / total << " %" << endl;
    cout << "Percentual de sapos: " << 100.0 * qtdSapos / total << " %" << endl;
        
    return 0;
}
